README.md

```diff
@@ -289,6 +301,214 @@ Gulp.js is used for building CSS assets.
+
+Changes and customizations of DataTables can generally be handled in `ckanext-querytool/ckanext/querytool/fanstatic/javascript/vitals.js` or `ckanext-querytool/ckanext/querytool/fanstatic/javascript/modules/table-module.js`.
+
+An example of custom changes can be found at the very top of [`vitals.js`](https://github.com/datopian/ckanext-querytool/blob/master/ckanext/querytool/fanstatic/javascript/vitals.js#L1-L273), where a feature plugin (`RowsGroup`) is used to automatically merge columns cells based on their values eqaulity.
+
+For more information on how to customize DataTables, see the [DataTables documentation](https://datatables.net/manual/).
+
+#### Leaflet
+
+This extension uses [Leaflet](https://leafletjs.com/) for map charts/visualizations. Leaflet is a JavaScript library for creating interactive maps, built on top of [OpenStreetMap](https://www.openstreetmap.org/).
+
+Changes and customizations of Leaflet can generally be handled in `ckanext-querytool/ckanext/querytool/fanstatic/javascript/modules/map-module.js`.
+
+#### General
+
+Most general jQuery and JavaScript customizations/overrides can be handled in `ckanext-querytool/ckanext/querytool/fanstatic/javascript/vitals.js`. Here's a small example where we watch for changes in the filter dropdowns and trigger a click on the "Update" button when a change is detected (code can be found [here](https://github.com/datopian/ckanext-querytool/blob/master/ckanext/querytool/fanstatic/javascript/vitals.js#L1304-L1306)):
+```
+$("body").on('change','.has-filter .filter-item-value', function() {
+  $('.btn-update').trigger("click");
+})
+```
+
+For more information on how to customize jQuery and JavaScript, see the [jQuery documentation](https://api.jquery.com/) and [JavaScript documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript).
+
+#### UI Options for Charts and Tables
+
+For most changes to the options found in the UI (for charts and tables, e.g. https://YOUR_SITE/report/edit_visualizations/REPORT_NAME), you can find the relevant code in the following files:
+
+- [viz-preview.js](https://github.com/datopian/ckanext-querytool/blob/master/ckanext/querytool/fanstatic/javascript/modules/viz-preview.js)
+- [visualizations_settings.js](https://github.com/datopian/ckanext-querytool/blob/master/ckanext/querytool/fanstatic/javascript/ckanext/querytool/fanstatic/javascript/visualizations_settings.js)
+
+#### HTML/Jinja2 Templates
+
+CKAN (and this extension) use [Jinja2](https://jinja.palletsprojects.com/en/3.1.x/) for templating. The templates for this extension can be found in [`ckanext-querytool/ckanext/querytool/templates`](https://github.com/datopian/ckanext-querytool/blob/master/ckanext/querytool/templates). Any changes in these HTML files should be reflected immediately in the UI immediately, but if you're not seeing them, you may need to restart CKAN. 
```

Comment:
_CKAN (and this extension) use [Jinja2](https://jinja.palletsprojects.com/en/3.1.x/) for templating_
Not 100% sure but I think it should be _uses_ ?